core/domain/classifier_domain_test.py

```diff
@@ -116,9 +356,19 @@ def test_validation(self):
             'Expected id to be a string')):
             training_job.validate()
 
+        # Verify validation error is raised when int is provided for
```

Comment:
Break this function into smaller functions each verifying a single type of validation error.

Reply:
The already implemented version already had checked for some of the cases in the same function so I just extended it. Furthermore, since we are using the same dictionary and just checking for each part of the dict, won't it be repetitive to initialize the dictionary again and again in each new function? Although, this function tests for multiple behaviours, each behaviour is very closely related to another so I think it would be better to write all the tests in the same function.  Also, all these tests for testing the same Validate() function so clubbing them together makes more sense according to me. What are your thoughts?

Reply:
> Furthermore, since we are using the same dictionary and just checking for each part of the dict, won't it be repetitive to initialize the dictionary again and again in each new function?

You do not need to initialize the dictionary multiple times - you can use a `setUp` function to have the initialization.

> Although, this function tests for multiple behaviours, each behaviour is very closely related to another so I think it would be better to write all the tests in the same function.

Though the behaviors are closely related, they are still different and failure of single behavior will result in the failure of complete validation test. On the other hand, if you have single behavior per function, a single test corresponding to the behavior will fail which is easier to debug and clearly shows the behavior which is not working correctly.

> Also, all these tests for testing the same Validate() function so clubbing them together makes more sense according to me.

The tests are for the same function which does multiple tasks - so each task should be tested separately. 

You can go through the rule 2 [here](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests) to get a better idea.

Do let me know if you have any further issues.

Thanks!